app/controllers/collections_controller.rb

```diff
@@ -0,0 +1,26 @@
+class CollectionsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @recipes = current_user.collection.recipes
+  end
+
+  def create
+    current_user.collection.recipes << current_recipe
+    current_user.save
+    flash[:notice] = "Recipe was added to collection"
+    redirect_to request.referrer
+  end
+
+  def destroy
+    current_user.collection.recipes.delete(current_recipe)
+    current_user.save
+    flash[:notice] = "Recipe was removed from collection"
+    redirect_to request.referrer
```

Comment:
Please look into:

```
redirect_back(fallback_location: root_path)
```

Same thing on LOC 12

Reply:
Also, I think I would prefer to see a notice to the user. Both on `destroy` and `create`

Reply:
There is a notice to the user on `destroy` and `create`